App/Containers/AddCaptionScreen.js

```diff
@@ -58,29 +59,45 @@ class AddCaptionScreen extends React.Component {
 
   componentWillMount () {
     this.props.navigation.setParams({
-      close: () => { this.props.close() },
+      close: this._close.bind(this),
       submit: () => { this._share(this.state.value) },
       submitEnabled: false
     })
   }
 
-  _share (comment) {
-    this.props.share(this.props.navigation.state.params.photo.id, [this.props.navigation.state.params.thread.id], comment)
-    this.props.close()
-    this.props.navigation.goBack()
+  _close () {
+    const params = this.props.navigation.state.params
+    if (params.photo) {
+      // from the wallet
+      this.props.close()
+    } else {
+      // from the photo picker
+      this.props.cancelShare(params.threadId, params.image)
```

Comment:
just allows this page to arise from sharing from Wallet or from sharing from cameraroll